Upgrade to PowSyBl dependencies 2022.0.0

OpenSuze opened the pull request:
Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
Add sensitivity batch size parameter
Move Xnec Selection outside of flow decomposition algorithm


**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*



**What is the current behavior?** *(You can also link to an open issue here)*



**What is the new behavior (if this is a feature change)?**



**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*


**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

OpenSuze committed:
Add sensi batch size parameter

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

OpenSuze committed:
Update CI

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

EtienneLt commented on java/pom.xml:
```diff
@@ -74,7 +74,7 @@
         <maven-compiler-plugin.version>3.10.1</maven-compiler-plugin.version>
         <maven-shade-plugin.version>3.2.4</maven-shade-plugin.version>
-        <powsybl-dependencies.version>1.3.1</powsybl-dependencies.version>
+        <powsybl-dependencies.version>1.4.0-SNAPSHOT</powsybl-dependencies.version>
```
could you use integration branch please ?

EtienneLt replied:
maybe it needs to be updated

OpenSuze replied:
rebased to integration branch :)

OpenSuze replied:
I removed this integration for CI tests

OpenSuze committed:
Fix typo

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

OpenSuze committed:
Fix CI Geoffroy

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

OpenSuze committed:
Remove Xnec Strategy selection
Add XnecProviderByIds support to python

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

murgeyseb commented on java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java:
```diff
@@ -0,0 +1,16 @@
+package com.powsybl.python.flow_decomposition;
```
Copyright header

murgeyseb commented on java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java:
```diff
@@ -0,0 +1,16 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
```
Author header

OpenSuze committed:
Remove Xnec Selection strategy
Add header

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

OpenSuze committed:
Moved xnec provider
Roll-back modified test

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

OpenSuze committed:
Fix pylint

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

OpenSuze committed:
Update doc :)

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

EtienneLt commented on tests/test_flow_decomposition.py:
```diff
@@ -93,6 +115,8 @@ def test_flow_decomposition_parameters():
     test_demo()
     print('\n\n\n\n')
+    test_demo_one_by_one()
+    print('\n\n\n\n')
```
a little thing could you remove all the print in this file ?

OpenSuze replied:
done

geofjamg committed:
Merge branch 'main' into add_parameter_sensi_batch_size

geofjamg committed:
Update to powsybl dependencies 2.0.0

Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@gmail.com>

geofjamg commented:
I upgraded to powsybl dependencies 2.0.0 and merge some of the modification of #499.
Some unit test are not passing because adaptation for security analysis status PR has not been done et still is to be done. The old isComputationOk is not working anymore and for pre contingency calculation always return ok... 
We need to adapt the code to get the status which is at the PreContingencyResult and PostContingencyResult class.

geofjamg commented:
@EtienneLt @sylvlecl is one of you available to work on contingency status?

EtienneLt commented:
> @EtienneLt @sylvlecl is one of you available to work on contingency status?

yes I can do that

EtienneLt commented on java/src/main/java/com/powsybl/python/flow_decomposition/FlowDecompositionContext.java:
```diff
@@ -0,0 +1,25 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
```
can you add the SPDX code
```java
/**
 * Copyright (c) 2022, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *  SPDX-License-Identifier: MPL-2.0
 */
```

EtienneLt committed:
adapt analysis results for powsybl core 5.0.0

Signed-off-by: Etienne LESOT <etienne.lesot@rte-france.com>

geofjamg commented on java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java:
```diff
@@ -164,14 +181,20 @@ private static void setSecurityAnalysisResultPointer(PyPowsyblApiHeader.Continge
 
-    private static PyPowsyblApiHeader.ArrayPointer<PyPowsyblApiHeader.ContingencyResultPointer> createContingencyResultArrayPointer(SecurityAnalysisResult result) {
+    private static PyPowsyblApiHeader.PreContingencyResultPointer createPreContingencyResultArrayPointer(SecurityAnalysisResult result) {
+        int resultCount = result.getPostContingencyResults().size() + 1; // + 1 for pre-contingency result
```
just one?

EtienneLt replied:
I forget to remove it now that they are separated

geofjamg commented on java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java:
```diff
@@ -138,14 +138,31 @@ public static void addContingency(IsolateThread thread, ObjectHandle contingency
+    private static void setPostContingencyResultInSecurityAnalysisResultPointer(PyPowsyblApiHeader.PostContingencyResultPointer contingencyPtr, PostContingencyResult postContingencyResult) {
+        contingencyPtr.setContingencyId(CTypeUtil.toCharPtr(postContingencyResult.getContingency().getId()));
+        contingencyPtr.setStatus(postContingencyResult.getStatus().ordinal());
+        List<LimitViolation> limitViolations = postContingencyResult.getLimitViolationsResult().getLimitViolations();
```
Maybe we could create a function to share limit violations code between pre and post contingency?

geofjamg committed:
Better use 2022.0.0 version

Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@gmail.com>

OpenSuze committed:
Update doc
Remove Network.create_aliases in network.rst
Remove prints from tests
Add SPDX

Signed-off-by: Hugo SCHINDLER <hugo.schindler@rte-france.com>

OpenSuze reviewed (commented):
Ready for merging. Everything looks ok on my side :) Thanks for the update !
